cpp/src/groupby/hash/groupby.cu

```diff
@@ -65,6 +65,15 @@ namespace detail {
 namespace hash {
 namespace {
 
+// TODO: replace it with `cuco::static_map`
+// https://github.com/rapidsai/cudf/issues/10401
+using map_type = concurrent_unordered_map<
+  cudf::size_type,
+  cudf::size_type,
+  cudf::experimental::row::hash::device_row_hasher<cudf::detail::default_hash,
```

Comment:
In stream compaction we use `compaction_hash`. Doesn't it support nested type?

Reply:
`compaction_hash` is just a wrapper of whatever hasher we use but adds sentinel handling. If the encapsulated hasher is `experimental::row::hash::device_row_hasher`, then yes, it supports nested type.